File: ios/RNZendrive/Source/RNZendriveInsurance.m
#import "React/RCTBridgeModule.h"
@interface RCT_EXTERN_MODULE(RCTZendriveInsurance, NSObject)
RCT_EXTERN_METHOD(
                  startDriveWithPeriod1: (RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject
                  )
RCT_EXTERN_METHOD(
                  startDriveWithPeriod2: (NSString *)trackingId
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject
                  )
RCT_EXTERN_METHOD(
                  startDriveWithPeriod3: (NSString *)trackingId
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject
                  )
RCT_EXTERN_METHOD(
                  stopPeriod: (RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject
                  )
@end

File: ios/RNZendrive/Source/RNZendriveInsurance.swift
//
//  RNZendriveInsurance.swift
//  RNZendrive
//
//  Created by Madhusudhan Sambojhu on 30/07/19.
//

import Foundation
import ZendriveSDK.Insurance
@objc(RCTZendriveInsurance)
class RNZendriveInsurance: NSObject {

    @objc
    func constantsToExport() -> [AnyHashable: Any]! {
        return [:]
    }

    @objc
    func startDriveWithPeriod1(
        _ resolve: @escaping RCTPromiseResolveBlock,
        rejecter reject: RCTPromiseRejectBlock
    ) {
        ZendriveInsurance.startDrive(period1: Utils.apiCallbackHandler(wtihResolver: resolve))
    }

    @objc
    func startDriveWithPeriod2(
        _ trackingId: String,
        resolver resolve: @escaping RCTPromiseResolveBlock,
        rejecter reject: RCTPromiseRejectBlock
    ) {
        ZendriveInsurance.startDrive(withPeriod2: trackingId,
                                     completionHandler: Utils.apiCallbackHandler(wtihResolver: resolve))
    }

    @objc
    func startDriveWithPeriod3(
        _ trackingId: String,
        resolver resolve: @escaping RCTPromiseResolveBlock,
        rejecter reject: RCTPromiseRejectBlock
    ) {
        ZendriveInsurance.startDrive(withPeriod3: trackingId,
                                     completionHandler: Utils.apiCallbackHandler(wtihResolver: resolve))
    }

    @objc
    func stopPeriod(
        _ resolve: @escaping RCTPromiseResolveBlock,
        rejecter reject: RCTPromiseRejectBlock
    ) {
        //        ZendriveInsurance.stopPeriod(Utils.apiCallbackHandler(wtihResolver: resolve))
        ZendriveInsurance.stopPeriod(Utils.insuranceApiCallbackHandler(wtihResolver: resolve))
    }

    @objc
    static func requiresMainQueueSetup() -> Bool {
        return false
    }
}
